srml/contract/src/wasm/env_def/macros.rs

```diff
@@ -36,19 +36,35 @@ macro_rules! gen_signature {
 	( ( $( $params: ty ),* ) -> $returns: ty ) => (
 		{
 			$crate::parity_wasm::elements::FunctionType::new(convert_args!($($params),*), Some({
-				use $crate::vm::env_def::ConvertibleToWasm; <$returns>::VALUE_TYPE
+				use $crate::wasm::env_def::ConvertibleToWasm; <$returns>::VALUE_TYPE
 			}))
 		}
 	);
 }
 
+#[macro_export]
+macro_rules! gen_signature_dispatch {
+	( $needle_name:ident, $needle_sig:ident ; $name:ident ( $ctx:ident $( , $names:ident : $params:ty )* ) $( -> $returns:ty )* , $($rest:tt)* ) => {
```

Comment:
can we wrap this or does that break macro parsing?

Reply:
Can be wrapped.